chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -207,55 +202,105 @@ void MdTextButton::SetLoading(bool loading) {
   UpdateColors();
 }
 
-void MdTextButton::UpdateBackgroundColor() {
-  // Handled via |UpdateColorsForBrave|.
-  if (kind_ != kOld) {
+void MdTextButton::UpdateTextColor() {
+  MdTextButtonBase::UpdateTextColor();
+
+  // Once we update the buttons across Brave to use the new style, we can remove
+  // this branch.
+  if (kind_ == kOld) {
+    if (GetProminent()) {
+      return;
+    }
+    const ui::NativeTheme* theme = GetNativeTheme();
+    // Override different text hover color
+    if (theme->GetPlatformHighContrastColorScheme() !=
+        ui::NativeTheme::PlatformHighContrastColorScheme::kDark) {
+      SetTextColor(ButtonState::STATE_HOVERED, kBraveBrandColor);
+      SetTextColor(ButtonState::STATE_PRESSED, kBraveBrandColor);
+    }
     return;
   }
 
-  MdTextButtonBase::UpdateBackgroundColor();
+  auto colors = GetButtonColors();
+  SetTextColor(GetVisualState(), colors.text_color);
 }
 
-void MdTextButton::UpdateOldColorsForBrave() {
-  if (GetProminent()) {
+void MdTextButton::UpdateBackgroundColor() {
+  // Once we update the buttons across Brave to use the new style, we can remove
+  // this branch.
+  if (kind_ == kOld) {
+    MdTextButtonBase::UpdateBackgroundColor();
+
+    // We don't modify the Prominent button at all.
+    if (GetProminent()) {
+      return;
+    }
+
+    // Override border color for hover on non-prominent
```

Comment:
I'm confused about these changes, why aren't we just overriding the ColorProvider?

Reply:
I assume that we had to copy the same code  and same bg color(`kColorDialogBackground`) so used `kBraveBrandedColor` directly instead of updating `kColorButtonBorder`. 

As we'll use `ui::ButtonStyle` instead of `Kind` from `cr126`, I think we could try to override this button style's color via color provider instead of using `GetButtonThemes()`.